### PR TITLE
[webgpu] Add case for matmul and conv2d to test special path

### PR DESCRIPTION
**Pull request by axinging**
Current yarn test doesn't test MatMulPackedVec4Program and Conv2DMMVec4Program (when remainder != 0). This PR adds better coverage.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5253)

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7 @haoyunfeix @xhcao @gyagp, PTAL

File: tfjs-backend-webgpu/src/conv2d_test.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import * as tf from '@tensorflow/tfjs-core';
+import {test_util} from '@tensorflow/tfjs-core';
+import {describeWebGPU} from './test_util';
+
+describeWebGPU('conv2d', () => {
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/conv2d_test.ts:22]: There is a `conv2d_webgpu_test.ts` file under src/kernels. Do you want to add the test there?

File: tfjs-backend-webgpu/src/matmul_test.ts [hunk 2]
```diff
@@ -877,4 +877,119 @@ describeWebGPU('matmul', () => {
...
+      31,  28,  29,  11, 11, 10, 7,  39,  48,  49,  37,  8,  15, 13, 7,  43,
+      54,  49,  18,  9,  16, 12, 7,  23,  31,  28,  29,  11, 11, 10, 7,  39,
+      47,  41,  23,  12, 15, 12, 8,  32,  41,  41,  42,  11, 14, 13, 8,  52,
+      64,  62,  31,  9,  19, 15, 8,  36,  47,  41,  23,  12, 15, 12, 8,  32,
+      29,  23,  27,  14, 11, 10, 8,  40,  49,  53,  52,  9,  16, 15, 8,  60,
+      74,  68,  17,  9,  21, 15, 8,  20,  29,  23,  27,  14, 11, 10, 8,  40
+    ]);
+  });
+
+  it('matMul MatMulPackedVec4Program outputShape[1] <= 1', async () => {
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/matmul_test.ts:974]: nit: `outputShape[1] = 1`. Can `outputShape[1]` be less than 1?

**Review by qjia7 — COMMENTED** [review]
LGTM with nits.

File: tfjs-backend-webgpu/src/conv2d_test.ts (hunk 1 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/conv2d_test.ts]: done

File: tfjs-backend-webgpu/src/matmul_test.ts (hunk 2 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/matmul_test.ts]: done

File: tfjs-backend-webgpu/src/kernels/conv2d_webgpu_test.ts [hunk 3]
```diff
@@ -163,7 +163,83 @@ describeWebGPU('im2col as separate shader', () => {
         tf.tensor4d(inputs.filter, [fSize, fSize, inputDepth, outputDepth]);
 
     const result = tf.conv2d(x, w, stride, pad);
-    tf.test_util.expectArraysClose(
+    test_util.expectArraysClose(
         await result.data(), [58.0, 78.0, 98.0, 118.0, 138.0, 158.0]);
   });
+
+  it('conv2d x=[1,8,8,3] f=[3,3,3,64] s=[2,2] d=1 p=valid Conv2DMMVec4Program remainder != 0',
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/conv2d_webgpu_test.ts:170]: You should add a new `describeWebGPU('conv2d',` in this file. The current one is for `describeWebGPU('im2col as separate shader', () => {
`.

Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/conv2d_webgpu_test.ts:171]: done

**Review by qjia7 — APPROVED** [review]
@pyu10055 @lina128 @jinjingforever take a look, thanks.

**Review by jinjingforever — APPROVED** [review]

**Pull request merged**